sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/CosmosClientBuilder.java

```diff
@@ -429,6 +430,15 @@ AzureKeyCredential getCredential() {
         return credential;
     }
 
+    /**
+     * Gets the {@link ProactiveContainerInitConfig} to be used
+     *
+     * @return {@link ProactiveContainerInitConfig}
+     * */
+    ProactiveContainerInitConfig getProactiveContainerInitConfig() {
```

Comment:
still feel it would be nice to include this config in the diagnostics.

Reply:
+100 :-)

Reply:
But IMO all of this can happen in subsequent PRs - first one is mostly to get public API surface area in

Reply:
Done